src/main/java/chatbot/ui/UI.java

```diff
@@ -0,0 +1,137 @@
+     */
+    public void listTaskOn(List<Task> tasks) {
+        if (tasks != null) {
+            System.out.println("\tThese are your tasks for that day");
+            list(tasks);
+        } else {
+            System.out.println("\tWell you are a lazy bum, you have nothing on the day");
+        }
+    }
+
+    /**
+     * Displays the tasks that contain a given keyword.
+     * @param tasks The list containing the relevant tasks.
+     */
+    public void listFound(List<Task> tasks) {
+        if (!tasks.isEmpty()) {
+            System.out.println("\tThese are the tasks that match your description");
+            list(tasks);
+        } else {
+            System.out.println("\t You like to search for nothing and waste time huh?");
+        }
+    }
+
+    private void list(List<Task> tasks) {
+        for (int i = 1; i <= tasks.size(); i++) {
+            System.out.println("\t" + i + ". " + tasks.get(i - 1));
+        }
+    }
+
+    /**
+     * Informs user that the given task has been added and the new todo list status.
+     *
+     * @param target The task added.
+     * @param numberOfTasks The number of tasks after the addition.
+     */
+    public void add(Task target, int numberOfTasks) {
+        System.out.println("\tLazily added this task for you " + EMOJI);
+        System.out.println("\t\t" + target);
+        System.out.println("\tWala now you have " + numberOfTasks + " tasks in the list.");
+    }
+
+    /**
+     * Informs user that the given task has been removed and the new todo list status.
+     *
+     * @param target The task removed.
+     * @param numberOfTasks The number of tasks after the deletion.
+     */
+    public void delete(Task target, int numberOfTasks) {
+        System.out.println("\tYES, I've removed this task for YOU:");
+        System.out.println("\t\t" + target);
+        System.out.println("\tWala now you have " + numberOfTasks + " tasks in the list.");
+    }
+
+    /**
+     * Informs user that the given task has been marked.
+     *
+     * @param target The task marked as complete.
+     */
+    public void unmark(Task target) {
+        System.out.println("\t-_-, I've unmarked this task for YOU AGAIN:");
+        System.out.println("\t\t" + target);
+    }
+
+    /**
+     * Informs user that the given task has been unmarked.
+     *
+     * @param target The task marked as incomplete.
+     */
+    public void mark(Task target) {
+        System.out.println("\tWellz, I've marked this task for YOU:");
+        System.out.println("\t\t" + target);
+    }
+
+    /**
+     * Displays the exception that has been raised.
+     *
+     * @param e The exceptions raised.
+     */
+    public void reprimand(DukeException e) {
+        System.out.println("\t" + e.getMessage());
+    }
+
+    /**
+     * Says goodbye to the user.
+     */
+    public void bye() {
+        System.out.println("\tBye. zzz FINALLY~~" + " " + EMOJI);
+        reader.close();
+    }
+}
```

Comment:
Generally well-written code with no major code quality issues! Keep up the good work!